### PR TITLE
Vpnaas: Create IKE policy

### PR DESCRIPTION
**Pull request by simonre**
For #723 

Links to the line numbers/files in the OpenStack source code that support the
code in this PR:

https://github.com/openstack/neutron-vpnaas/blob/058469e1b99b647537a5228c6a384d93df5484df/neutron_vpnaas/db/vpn/vpn_db.py#L312

API:

https://developer.openstack.org/api-ref/network/v2/#create-ike-policy

### CODE REVIEW

**Comment by simonre** [comment]
@jtopjian This is ready for review

File: openstack/networking/v2/extensions/vpnaas/ikepolicies/requests.go [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+	// AuthAlgorithm is the authentication hash algorithm.
+	// Valid values are sha1, sha256, sha384, sha512.
+	// The default is sha1.
+	AuthAlgorithm AuthAlgorithm `json:"auth_algorithm,omitempty"`
+
+	// EncapsulationMode is the encapsulation mode.
+	// A valid value is tunnel or transport.
+	// Default is tunnel.
+	EncapsulationMode EncapsulationMode `json:"encapsulation_mode,omitempty"`
```
Comment (jtopjian) [inline openstack/networking/v2/extensions/vpnaas/ikepolicies/requests.go]: It doesn't look like this is a valid field for IKE?

File: acceptance/openstack/networking/v2/extensions/vpnaas/ikepolicy_test.go [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+package vpnaas
+
+import (
+	"testing"
+
+	"github.com/gophercloud/gophercloud/acceptance/clients"
+	"github.com/gophercloud/gophercloud/acceptance/tools"
+)
+
+func TestPolicyCRUD(t *testing.T) {
```
Comment (jtopjian) [inline acceptance/openstack/networking/v2/extensions/vpnaas/ikepolicy_test.go]: As mentioned in the IPSec PR, this test name clashes so we'll want to change it.

File: openstack/networking/v2/extensions/vpnaas/ikepolicies/requests.go [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+	// PFS is the Perfect forward secrecy mode.
+	// A valid value is Group2, Group5, Group14, and so on.
+	// Default is Group5.
+	PFS PFS `json:"pfs,omitempty"`
+
+	// TransformProtocol is the transform protocol.
+	// A valid value is ESP, AH, or AH- ESP.
+	// Default is ESP.
+	TransformProtocol TransformProtocol `json:"transform_protocol,omitempty"`
```
Comment (jtopjian) [inline openstack/networking/v2/extensions/vpnaas/ikepolicies/requests.go]: It doesn't look like this is a valid field, either?

File: openstack/networking/v2/extensions/vpnaas/ikepolicies/results.go [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+	Lifetime Lifetime `json:"lifetime"`
+
+	// ID is the ID of the policy
+	ID string `json:"id"`
+
+	// Phase1NegotiationMode is the IKE mode
+	Phase1NegotiationMode string `json:"phase1_negotiation_mode"`
+
+	// IkeVersion is the IKE version.
+	IkeVersion string `json:"ike_version"`
```
Comment (jtopjian) [inline openstack/networking/v2/extensions/vpnaas/ikepolicies/results.go]: Let's rename this to `IKEVersion`.

**Comment by simonre** [comment]
@jtopjian I've rebased this and addressed your comments. I think this is ready for re-review

File: acceptance/openstack/networking/v2/extensions/vpnaas/vpnaas.go [hunk 5]
```diff
@@ -63,6 +64,30 @@ func CreateIPSecPolicy(t *testing.T, client *gophercloud.ServiceClient) (*ipsecp
 	}
 
 	t.Logf("Successfully created IPSec policy %s", policyName)
+	t.Logf("Successfully created IKE policy %s", policyName)
```
Comment (jtopjian) [inline acceptance/openstack/networking/v2/extensions/vpnaas/vpnaas.go]: Really minor nit, but looks like this line should be removed.

**Comment by simonre** [comment]
The projectID field is returned by my working environment. I added it to the result and changed the unit tests.

**Comment by jtopjian** [comment]
@simonre Thank you for checking into that!

This looks good. Nice work on this. Thank you again for all of your patience and work on this suite.

**Pull request merged**